### ISSUE TITLE
kwargs for `wrap` in `TextureMap`?

### ISSUE THREAD

**Issue by kushalkolar** [pygfx/pygfx#913]
Hi! Catching up with #889 , and I saw [this example](https://docs.pygfx.org/latest/_gallery/validation/validate_image_colormap.html) and wondering about the various options for `warp` which seem to correspond to the wgpu [`AddressMode` enums](https://wgpu.rs/doc/wgpu/enum.AddressMode.html) . I don't see `"clamp"`, is that an alias for `"clamp-to-border"`? Both seem to produce the same effect if I slightly modify the example:

For scientific visualizations using colormaps for images I'm assuming that `clamp-to-border` is usually what people want? 

```python
import numpy as np
from wgpu.gui.auto import WgpuCanvas, run
import pygfx as gfx


renderer = gfx.renderers.WgpuRenderer(WgpuCanvas(size=(600, 600)))

im = np.repeat(np.linspace(0, 2, 100).reshape(1, -1), 48, 0).astype(np.float32)
colormap = gfx.Texture(np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]], np.float32), dim=1)

image = gfx.Image(
    gfx.Geometry(grid=gfx.Texture(im, dim=2)),
    gfx.ImageBasicMaterial(
        clim=(0, 1),
        map=gfx.TextureMap(
            colormap,
            filter="linear",
            wrap="clamp",
        ),
    ),
)
image.local.y = 52

scene = gfx.Scene()
scene.add(image)

camera = gfx.OrthographicCamera()
camera.show_rect(-4, 103, -4, 103)

renderer.request_draw(lambda: renderer.render(scene, camera))


if __name__ == "__main__":
    run()
```

![image](https://github.com/user-attachments/assets/4d35c489-1dde-4fe1-9657-44860e56f3d5)

Thanks!

**Comment by Vipitis** [comment]
colormaps are like one pixel thin, so kinda making them one dimensional. In this case all the address modes will behave nearly identical. Clamp to edge is the default and the conversion happens here: https://github.com/pygfx/pygfx/blob/4844ea0d030ef5b1fde8001aa1ff6e8721b976be/pygfx/renderers/wgpu/engine/update.py#L194

What might be interesting is that you can chose a mode for each texture direction. which could be useful for clipped values in a color map? The features for clip to zero aren't implemented in wgpu-native yet.

**Comment by panxinmiao** [comment]
"Wrap" mode is used to control the sampling behavior of texture coordinates (UV) when they go beyond the [0, 1] range. It usually comes into play when textures are mapped to more complex geometries. When the UV coordinates exceed the [0, 1] range, wrap mode determines how they are “wrapped” back into the valid range (for example, using REPEAT or CLAMP_TO_EDGE).

In the case of a gfx.Image object, the geometry is a plane, and the UV coordinates are already confined within the [0, 1] range, so wrap mode does not affect the texture sampling behavior. In other words, wrap mode mainly takes effect when the texture is mapped to geometries where the UV coordinates go beyond the [0, 1] range.

**Comment by kushalkolar** [comment]
Thanks for the details! Even for images all the modes do not behave identical, in the example above if we change the mode to `"repeat"` we get this:

![image](https://github.com/user-attachments/assets/5a81ef16-729f-452d-9caf-0e4cbd5c5dcf)

and for `"mirror-repeat"`:

![image](https://github.com/user-attachments/assets/c2a09326-4075-464f-9fec-55d5b8cee59f)

Anyways I think for scientific images it makes most sense to keep it at `"clamp-to-edge"`.

**Comment by panxinmiao** [comment]
Ah, it seems that the UV of the Image is not restricted to the [0, 1] range as I originally thought.

Currently, the engine handles TextureMap with a unified logic, following the behavior defined by the TextureMap.

> Anyways, I think for scientific images, it makes most sense to keep it at 'clamp-to-edge.'

If that's the case, since the map property of a Material can accept both Texture and TextureMap, we can set the wrap mode to "clamp" by default when passing in a Texture. If a TextureMap is passed in, and the wrap mode is explicitly defined, my suggestion would be to follow the wrap settings of the TextureMap (perhaps with a warning if it's not set to "clamp"?), as this could very well be the user's intended behavior.

**Comment by kushalkolar** [comment]
> Ah, it seems that the UV of the Image is not restricted to the [0, 1] range as I originally thought.

Yes, it does make sense for it to extend beyond the range, for example if you are displaying with a distribution that is highly tailed you may want to set the max clim much lower than the data max. 

> If that's the case, since the map property of a Material can accept both Texture and TextureMap, we can set the wrap mode to "clamp" by default when passing in a Texture. If a TextureMap is passed in, and the wrap mode is explicitly defined, my suggestion would be to follow the wrap settings of the TextureMap (perhaps with a warning if it's not set to "clamp"?), as this could very well be the user's intended behavior.

Sounds like a good idea.

**Comment by almarklein** [comment]
> Ah, it seems that the UV of the Image is not restricted to the [0, 1] range as I originally thought.

It is, but the values that it has that sample into the colormap are not. That's what @kushalkolar's example shows.

> we can set the wrap mode to "clamp" by default when passing in a Texture.

I think that would indeed be a good/better default.

**Comment by almarklein** [comment]
Closed by #918
